client/app/components/Input/InputTextarea.jsx

```diff
@@ -67,84 +67,75 @@ export type State = {
   value?: any,
 };
 
-export class InputTextarea extends React.Component<Props, State> {
-  editor: any;
-
-  editorRef: any;
-
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      value: props.value || '',
-    };
-  }
-
-  componentDidMount() {
-    const { value } = this.state;
-    if (this.editorRef) {
-      this.editor = init({
-        element: this.editorRef.getElementsByClassName('editor')[0],
-        onChange: this.onChange,
-        classes,
-        actions,
-      });
-      this.editor.content.innerHTML = value;
-    }
-  }
-
-  onChange = (value: string) => {
-    this.setState({ value });
+export function InputTextarea({
+  id,
+  name,
+  value,
+  required,
+  hasError,
+  myRef,
+  dark,
+}: Props) {
+  const [currentValue, setValue] = useState<State>(value || '');
+  let editorRef;
+  let editor;
+
+  const onChange = (updatedValue: string) => {
+    setValue(updatedValue);
   };
 
-  onBlur = () => {
-    const { required, hasError } = this.props;
-    const { value } = this.state;
+  const onBlur = () => {
     if (required && hasError) {
-      hasError(!value || value === '<p><br></p>');
+      hasError(!currentValue || currentValue === '<p><br></p>');
     }
   };
 
-  onFocus = () => {
-    const { required, hasError } = this.props;
+  const onFocus = () => {
     if (required && hasError) {
       hasError(false);
     }
-    if (this.editorRef) {
-      this.editorRef.getElementsByClassName('editorContent')[0].focus();
+    if (editorRef) {
+      editorRef.getElementsByClassName('editorContent')[0].focus();
     }
   };
 
-  displayHidden = () => {
-    const { name, required, myRef } = this.props;
-    const { value } = this.state;
-    return (
-      <input
-        type="hidden"
-        value={value}
-        name={name}
-        required={required}
-        ref={myRef}
-      />
-    );
-  };
-
-  render() {
-    const { id, dark } = this.props;
-    return (
-      <div
-        id={id}
-        className={`${inputCss.default} ${dark ? css.dark : ''}`}
-        onBlur={this.onBlur}
-        onFocus={this.onFocus}
-        tabIndex={0}
-        role="textbox"
-        ref={(element) => {
-          this.editorRef = element;
-        }}
-      >
-        <div className={`editor ${css.editor}`} />
-        {this.displayHidden()}
-      </div>
-    );
-  }
+  const displayHidden = () => (
+    <input
+      type="hidden"
+      value={currentValue}
+      name={name}
+      required={required}
+      ref={myRef}
+    />
+  );
+
+  useEffect(() => {
+    if (editorRef) {
+      editor = init({
```

Comment:
Defining `editor` inside `useEffect()` (after a dependency list is added) will cause `editor` to be `undefined` in subsequent renders, since if `useEffect` only runs on mount then this definition will be skipped and the newly defined instance will be inaccessible outside this function scope.

One [option](https://reactjs.org/docs/hooks-reference.html#useref) is to `useRef` to store the value:

>However, `useRef()` is useful for more than the `ref` attribute. It’s [handy for keeping any mutable value around](https://reactjs.org/docs/hooks-faq.html#is-there-something-like-instance-variables) similar to how you’d use instance fields in classes.

```
const editor = useRef(null);
...
useEffect(() => {
  ...
  editor.current = init({ ... });
  editor.current.content.innerHTML = currentValue;
}, []);
```

Reply:
Two questions on this:
* All this code is doing is initializing Pell, attaching it to the ref passed by React, passing in methods to deal with changes, and then filling it with the starting content right? Wouldn't this only ever need to be done when the component is initialized? Why would we need to access the editor variable outside of that scope, and on re-render? Or, is this just so that future people have access if they want it?
* If we do want to keep the reference to editor alive past the initial render, wouldn't it make more sense to keep it around as a state variable rather than useRef? If we were for some reason to change the value of editor later on wouldn't we want to trigger a re-render of the component?

Reply:
>Wouldn't this only ever need to be done when the component is initialized? Why would we need access to the editor variable outside the scope and on re-render? Or, is this just so that future people have access if they want it?

Right, definitely! I think it makes sense to store the `editor` outside the callback, for extending editor-derived functionality (and also because it is analogous to storing a value on an instance field in a class component, i.e. `this.editor` becomes `editor.current`).

But currently it would also work _not_ to assign the editor to a variable at all, since it isn't being used anywhere else. I think the original implementation stored it on the class as a way to keep it handy if needed.

>as a state variable rather than useRef? If we were for some reason to change the value of editor later on wouldn't we want to trigger a re-render of the component?

I think it makes sense to wait for that use case to become clearer (since there are different possible paths for the implementation to go). For a more dynamic handling of the `editor`, it may become necessary to use a different approach at that point. For now, updating the `value` in state is enough for rendering the hidden input.

I guess it goes back to `useRef` being analogous to an instance field. Would we store `editor` in the state if it were a class component? Since the `editor` is currently autonomous and doesn't directly affect what the component renders, I think it can remain a static value 👍

Reply:
This is a good explanation, I agree. I'll make the change.